src/main/commandServer/__tests__/httpCommandServer.spec.ts

```diff
@@ -0,0 +1,68 @@
+import fs from 'fs';
+import os from 'os';
+import path from 'path';
+import { execFileSync } from 'child_process';
+
+import { HttpCommandServer } from '../httpCommandServer';
+import { spawnFile } from '@/utils/childProcess';
+
+describe(HttpCommandServer, () => {
+  let itWindows = it;
+  let itNonWindows = it;
+  let rdctlPath = path.join('resources', os.platform(), 'bin', 'rdctl');
+
+  if (os.platform().startsWith('win')) {
+    rdctlPath += '.exe';
+    itNonWindows = it.skip;
+    // Don't run the error-message test if the rancher-desktop WSL is running
+    if (execFileSync('wsl', ['--list', '--quiet'], { stdio: 'pipe', encoding: 'utf16le' }).match(/^rancher-desktop$/m) &&
+      execFileSync('wsl', ['--list', '--verbose'], { stdio: 'pipe', encoding: 'utf16le' }).match(/^\s*rancher-desktop\s+Running/m)) {
+      itWindows = it.skip;
+    }
+  } else {
+    itWindows = it.skip;
+    try {
+      execFileSync(rdctlPath, ['list-settings'], { stdio: 'pipe' });
+      itNonWindows = it.skip;
+    } catch {
+      // Run this again in the test to verify we get the expected error message
+    }
+  }
+  try {
+    fs.accessSync(rdctlPath, fs.constants.X_OK);
+  } catch (e: any) {
+    if (e.code === 'ENOENT') {
+      itWindows = itNonWindows = it.skip;
+    } else {
+      throw e;
+    }
+  }
+  /**
+   * This test is designed to handle two cases:
+   * 1. VM 0 doesn't exist (so rancher desktop isn't running, or never has).
+   * 2. Rancher Desktop isn't running.
+   * There are edge cases where this test might fail, such as when Rancher Desktop is starting up
+   * so the VM exists but the command server hasn't started yet. For the purposes of running this in CI,
+   * or by developers during a typical edit-test-fix cycle, these are edge cases we can ignore for now.
+   */
+  itNonWindows("should fail to run rdctl shell when server isn't running", async() => {
+    const listSettingsRejects = await expect(() => spawnFile(rdctlPath, ['list-settings'], { stdio: 'pipe' })).rejects;
```

Comment:
```suggestion
    const listSettingsRejects = expect(() => spawnFile(rdctlPath, ['list-settings'], { stdio: 'pipe' })).rejects;
```
We shouldn't await on rejects (because it's not a promise; and if it were, we wouldn't want the result before we have an assertion).

This applies to the two cases below too.

Reply:
Do you have a link to some docs that explains the syntax? I found the jest docs that showed what it is, but not how it works.

Reply:
Err, `.rejects` is not a promise, so `await`ing on it doesn't make much sense?